### DISCUSSION TITLE
Using tooltips without tooltip provider

### DISCUSSION THREAD

**Discussion by brianlovin** [radix-ui/primitives#2183] [Help] [answered]
Hey team — I was wondering if using the Tooltip component without needing a global Tooltip Provider wrapper is possible. Our app is set up such that we have a few different ways that our pages can be wrapped in providers and we keep encountering whack-a-mole situations where tooltips will occasionally be used on a component that isn't nested in a global provider wrapper.

Is it possible to use tooltips without the provider?

Alternatively, is there a major drawback to simply including the `<Tooltip.Provider>` wrapper in our `Tooltip.tsx` component, being used on every instance of a tooltip in our app?

**Comment by benoitgrelard** [comment, marked as answer]
Hey Brian,

> Is it possible to use tooltips without the provider?

No, it isn't possible to use without the provider. The provider lets you configure things like timings globally for your app, but also manages the ability to move from an element with a tooltip to another one with another tooltip without having to incur the delay again.

> Alternatively, is there a major drawback to simply including the <Tooltip.Provider> wrapper in our Tooltip.tsx component, being used on every instance of a tooltip in our app?

If having a global provider isn't possible for your, then yes you may do this, wrap your implemention with the provider, everything will work except the latter point mentioned above, meaning if 2 buttons with tooltips are next to each other, the user will still have to wait for the second one to open even if the first is open already and they move quickly to it.

Hope this helps.

✌️

**Reply by brianlovin** [reply]
Perfect, thank you for the explanation! We have a delay of 0, so we're not worried about the timings as users hover over different elements.

**Reply by clicktodev** [reply]
I actually have focus bugs when using the provider globally that i don't get when using a separate provider for each tooltip

**Reply by SansPapyrus683** [reply]
Can confirm. Using a global provider makes is so that moving from one tooltip trigger to another very quickly doesn't change the tooltip sometimes.

**Comment by pdrbrnd** [comment]
Hi @benoitgrelard I know this is an old discussion but I feel this is still something that wasn't completely answered.

While I definitely understand the value of the Provider, I still can't understand why should it throw when not present, forcing us to wrap storybook, tests, and everything with a Provider that, in those cases, is pretty much inconsequential.

If you feel that this is a design decision and shouldn't be changed in any way, at least you could expose the context for us to check in our implementation if there's a context available and wrap our Tooltip in its own provider if not.

Would this be an acceptable change?

**Reply by pdrbrnd** [reply]
I also see that radix's `createContext` [is prepared to accept a `defaultContext`](https://github.com/radix-ui/primitives/blob/main/packages/react/context/src/createContext.tsx#L20). Isn't this something that could be leveraged [within Tooltip](https://github.com/radix-ui/primitives/blob/main/packages/react/tooltip/src/Tooltip.tsx#L43) to make it non-mandatory?

**Comment by theskillwithin** [comment]
I can't seem to find any docs on setting up the Provider
